src/material/slider/slider-interface.ts

```diff
@@ -0,0 +1,215 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {InjectionToken} from '@angular/core';
+import {Thumb} from '@material/slider';
+import {MatRipple, RippleGlobalOptions} from '@angular/material/core';
+
+/**
+ * Injection token that can be used for a `MatSlider` to provide itself as a
+ * parent to the `MatSliderThumb` and `MatSliderRangeThumb`.
+ * Used primarily to avoid circular imports.
+ * @docs-private
+ */
+export const MAT_SLIDER_TOKEN = new InjectionToken<{}>('MAT_SLIDER_TOKEN');
+
+/**
+ * Injection token that can be used to query for a `MatSliderThumb`.
+ * Used primarily to avoid circular imports.
+ * @docs-private
+ */
+export const MAT_SLIDER_THUMB_TOKEN = new InjectionToken<{}>('MAT_SLIDER_THUMB_TOKEN');
+
+/**
+ * Injection token that can be used to query for a `MatSliderRangeThumb`.
+ * Used primarily to avoid circular imports.
+ * @docs-private
+ */
+export const MAT_SLIDER_RANGE_THUMB_TOKEN = new InjectionToken<{}>('MAT_SLIDER_RANGE_THUMB_TOKEN');
+
+/**
+ * Injection token that can be used to query for a `MatSliderVisualThumb`.
+ * Used primarily to avoid circular imports.
+ * @docs-private
+ */
+export const MAT_SLIDER_VISUAL_THUMB_TOKEN = new InjectionToken<{}>(
+  'MAT_SLIDER_VISUAL_THUMB_TOKEN',
+);
+
+export interface MatSliderInterface {
```

Comment:
Let's drop the `Interface` suffix from all of these and prefix them with an underscore to indicate that they're private. E.g. `_MatSlider`.